src/tooling/piral-cli-webpack5/src/helpers.ts

```diff
@@ -2,19 +2,31 @@ import { existsSync } from 'fs';
 import { Configuration } from 'webpack';
 import { DefaultConfiguration } from './webpack/common';
 
-export function extendConfig(
+export async function extendConfig(
   [webPackConfig, enhancer]: DefaultConfiguration,
   otherConfigPath: string,
   overrides: Configuration = {},
-): Configuration {
+): Promise<Configuration> {
   const original = webPackConfig;
 
   if (existsSync(otherConfigPath)) {
     try {
-      const otherConfig = require(otherConfigPath);
+      let otherConfig = require(otherConfigPath);
+      if (otherConfig.default) {
+        // The webpack config file appears to be an ESM module;
+        // this interop should give us the actual exported config
+        otherConfig = otherConfig.default;
+      }
 
       if (typeof otherConfig === 'function') {
-        webPackConfig = otherConfig(webPackConfig);
+        const configResult = otherConfig(webPackConfig);
+        // support Promise for returned config:
+        // https://webpack.js.org/configuration/configuration-types/#exporting-a-promise
+        if (configResult instanceof Promise) {
+          webPackConfig = await configResult;
```

Comment:
I think this is unnecessary. We can just `await configResult` independent if its a `Promise` or not. JavaScript `await` is lax here, it will just work:

```js
await 5; // 5
await Promise.resolve(5); // 5
await thisPromiseYieldsFive; // 5
```

is all equivalent; so no need to have the `instanceof` check.

Reply:
Sure I see the intention - I just want to make sure the code remains easy to digest and does not do unnecessary detours.

Also just a remark: The CLA was not signed as the associated email address does not seem to be registered with your account. You'll either need to add this email address or re-do commits with a different config using an email that you either have or can associate with your GitHub account.

Reply:
Any news on this one? Otherwise I'd close and re-do.